app/services/participants/build_completion_candidate_list.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# This service builds a list of candidates to check for an induction completion date
+# The list will be processed by the SetParticipantCompletionDateJob
+# This could be improved later to more intelligently select which records to include
+# as at the moment it's just including all 2021 and 2022 ECTs
+module Participants
+  class BuildCompletionCandidateList < BaseService
+    def call
+      ActiveRecord::Base.transaction do
+        clean_candidate_list
+        build_candidate_list
+      end
+      CompletionCandidate.count
+    end
+
+  private
+
+    def clean_candidate_list
+      CompletionCandidate.delete_all
+    end
+
+    def build_candidate_list
+      ActiveRecord::Base.connection.execute(
+        <<~SQL,
+          INSERT INTO completion_candidates (participant_profile_id)
+          SELECT pp.id
+          FROM participant_profiles pp
```

Comment:
Do we need to add the with an eligible status here as well as teacher profiles requiring trns as the old query above did?

Reply:
I don't think it matters, I added the eligible status before because I was trying to only do what was necessary but I don't think it makes any difference, if they have completed we should record that I think?  I will check with Nicola though, thanks.

Reply:
Confirmed that PO/BA would like to ignore eligibility for this

Reply:
I am excluding teacher profiles without TRNs, it's hidden under the SQL Injection warning

Reply:
ah gotcha, 👍